File: 0605-can-place-flowers/0605-can-place-flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        cnt = 0
        if n == 0:
            return True
        if len(flowerbed) == 1:
            return False if flowerbed[0] else True
        if flowerbed[0] == 0 and flowerbed[1] == 0:
            cnt += 1
        for i in range(1, len(flowerbed)-1):
            if flowerbed[i-1] == 0 and flowerbed[i] == 0 and flowerbed[i+1] == 0:
                print("i: ", i)
                flowerbed[i] = 1
                cnt += 1
        if flowerbed[-2] == 0 and flowerbed[-1] == 0:
            cnt += 1
        print("cnt: ", cnt)
        if cnt >= n:
            return True
        else:
            return False

File: 0605-can-place-flowers/0605-can-place-flowers.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        int count = 0;
        int m = flowerbed.size();
        
        if (m == 1) {
            if (flowerbed[0] == 0) count++;
            return n <= count;
        }

        if (flowerbed[0] == 0 && flowerbed[1] == 0) {
            flowerbed[0] = 1;
            count++;
        }
        for (int i = 1; i < flowerbed.size() - 1; i++) {
            if (flowerbed[i - 1] == 0 && flowerbed[i] == 0 && flowerbed[i + 1] == 0) {
                flowerbed[i] = 1;
                count++;
            } 
        }
        if (flowerbed[m - 1] == 0 && flowerbed[m - 2] == 0) {
            count++;
        }
        //cout << count << endl;
        //for (int flower: flowerbed) cout << flower << endl;
        return n <= count;
    }
};